sdk/nodejs/runtime/stack.ts

```diff
@@ -176,7 +176,8 @@ async function massageComplex(prop: any, objectStack: any[]): Promise<any> {
     if (Resource.isInstance(prop)) {
         // Emit a resource as a normal pojo.  But filter out all our internal properties so that
         // they don't clutter the display/checkpoint with values not relevant to the application.
-        return serializeAllKeys(n => !n.startsWith("__"));
+        const pojo = await serializeAllKeys(n => !n.startsWith("__"));
+        return !isDryRun() ? pojo : { ...pojo, "@isPulumiResource": true };
```

Comment:
i have zero issue with this approach.  however, i think it woudl be worthwhile to comment why we're only doing this in preview.  it's not immediately clear on reading this why this code is doing what it does.

Reply:
Good point; will do.